shell/platform/linux/fl_mouse_cursor_plugin.cc

```diff
@@ -0,0 +1,126 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/fl_mouse_cursor_plugin.h"
+
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_method_channel.h"
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_standard_method_codec.h"
+
+#include <gtk/gtk.h>
+
+static constexpr char kChannelName[] = "flutter/mousecursor";
+static constexpr char kBadArgumentsError[] = "Bad Arguments";
+static constexpr char kActivateSystemCursorMethod[] = "activateSystemCursor";
+static constexpr char kKindKey[] = "kind";
+
+struct _FlMouseCursorPlugin {
+  GObject parent_instance;
+
+  FlMethodChannel* channel;
+
+  FlView* view;
```

Comment:
This will eventually need to become an engine reference that you can query for the view.

Reply:
Yes, but I figure since all this API is internal we can change it later.